e2etests/web/regular_integration_tests/test_driver/text_editing_integration.dart

```diff
@@ -177,6 +177,21 @@ void main() {
       'cancelable': true,
     });
 
+    // Press and release AltGr key.
+    // Regression test for https://github.com/flutter/flutter/issues/58979.
+    dispatchKeyboardEvent(input, 'keydown', <String, dynamic>{
+      'key': 'AltGraph',
+      'code': 'AltRight',
+      'bubbles': true,
+      'cancelable': true,
+    });
+    dispatchKeyboardEvent(input, 'keyup', <String, dynamic>{
+      'key': 'AltGraph',
+      'code': 'AltRight',
+      'bubbles': true,
+      'cancelable': true,
+    });
```

Comment:
I think what we need here is a test to make sure that `metaState` is computed correctly when `AltGr` is pressed.

Reply:
I added this test to the framework side as well (after the flutter drive breakage) last week.

If this is a functionality that can break with framework code change, I suggest we also make the same change in file:  https://github.com/flutter/flutter/blob/master/dev/integration_tests/web_e2e_tests/test_driver/text_editing_integration.dart